delivery_carrier_return_barcode_pattern/models/delivery_carrier.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+import re
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+HELP_RETURN_BARCODE_PATTERN = """
+    This pattern can be used to parse a return barcode from the carrier.
+    This has to be a valid regular expression.
+
+    Example for barcode containing only the PO: (?P<origin>.*)
```

Comment:
My bad, it's SO and not PO
```suggestion
    Example for barcode containing only the sales order reference: (?P<origin>.*)
```